common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -3975,6 +3975,16 @@
     ],
     "sqlState" : "21000"
   },
+  "SCALAR_FUNCTION_NOT_COMPATIBLE" : {
+    "message" : [
+      "Cannot find a compatible ScalarFunction#produceResult"
+    ]
+  },
+  "SCALAR_FUNCTION_NOT_FULLY_IMPLEMENTED" : {
+    "message" : [
+      "ScalarFunction <scalarFunc> neither implements magic method nor override 'produceResult'"
```

Comment:
I think we should explicitly say what we expect: a method with the name `produceResult` which accepts a parameter of the type `InternalRow`

Reply:
Replaced with 
```
"ScalarFunction <scalarFunc> not implements or overrides method 'produceResult(InternalRow)'."
```